packages/tpetra/core/src/Tpetra_Map_def.hpp

```diff
@@ -1645,6 +1645,7 @@ namespace Tpetra {
         Kokkos::RangePolicy<LO, typename DT::execution_space>
           range (static_cast<LO> (0), static_cast<LO> (lgMap.size ()));
         Kokkos::parallel_for (range, *this);
+        Kokkos::fence();
```

Comment:
Again, it would be nicer to put the fence as close to the erroneous code as possible.  The code here is fine; the code that follows (probably that uses the local-to-global Map on host) is wrong.